### PR TITLE
CI: Switch to GitHub Actions and Codecov

### PR DESCRIPTION
**Pull request by alyst**
- drop *.travis.yml*
- add ubuntu-1-x86 testing
- add win-1-x64 testing

### CODE REVIEW

**Comment by alyst** [comment]
@nalimilan Could you please take a look if this PR looks fine? (In particular, conforms to the recent guidelines for using gh-actions for CI)

File: README.md [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 Methods for data clustering and evaluation of clustering quality.
 
-[![Travis](https://travis-ci.org/JuliaStats/Clustering.jl.svg?branch=master)](https://travis-ci.org/JuliaStats/Clustering.jl)
-[![Coveralls](https://coveralls.io/repos/github/JuliaStats/Clustering.jl/badge.svg?branch=master)](https://coveralls.io/github/JuliaStats/Clustering.jl?branch=master)
+[![Build Status](https://github.com/JuliaStats/Clustering.jl/workflows/CI/badge.svg)](https://github.com/JuliaStats/Clustering.jl/actions)
```
Comment (nalimilan) [inline README.md:5]:
```suggestion
[![Build Status](https://github.com/JuliaStats/Clustering.jl/workflows/CI/badge.svg)](https://github.com/JuliaStats/Clustering.jl/actions?query=workflow%3ACI+branch%3Amaster)
```

File: test/dbscan.jl [hunk 2]
```diff
@@ -56,14 +56,17 @@ end
 
     points = [p0 p1 p2]
 
+    # FIXME The current test implementation relies on random numbers sequence being exactly the same accross all tested julia versions,
+    #       so currently it's broken. We need a new test that is not (so much) seed-dependent.
```
Comment (nalimilan) [inline test/dbscan.jl:60]: How about using StableRNGs? I'm afraid if we skip these test we will forget to reenable them forever. :-/

**Review by nalimilan — APPROVED** [review]

File: test/dbscan.jl (hunk 2 above)
Comment (alyst) [reply test/dbscan.jl]: Thanks, StableRNGs were not on my radar.

**Pull request merged**